HPXMLtoOpenStudio/resources/weather.rb

```diff
@@ -310,14 +311,67 @@ def calc_ground_temperatures
     phi = Math::atan(z)
     bo = (data.MonthlyAvgDrybulbs.max - data.MonthlyAvgDrybulbs.min) * 0.5
 
-    # The regression coefficients are from L. Xing's simplified design model ground temperatures (Appendix A-3) and the nearest TMY3 station's annual dry bulb.
-    data.GroundAnnualTemp = UnitConversions.convert(0.91 * UnitConversions.convert(data.AnnualAvgDrybulb, 'F', 'C') + 1.82, 'C', 'F')
-
-    data.GroundMonthlyTemps = []
+    data.ShallowGroundMonthlyTemps = []
     for i in 0..11
       theta = amon[i] * 24.0
-      data.GroundMonthlyTemps << data.GroundAnnualTemp - bo * Math::cos(2.0 * Math::PI / p * theta - po - phi) * gm
+      data.ShallowGroundMonthlyTemps << UnitConversions.convert(data.AnnualAvgDrybulb - bo * Math::cos(2.0 * Math::PI / p * theta - po - phi) * gm + 460.0, 'R', 'F')
     end
+    data.ShallowGroundAnnualTemp = data.AnnualAvgDrybulb
+
+    if header.Latitude < 0
+      # Southern hemisphere
+      data.ShallowGroundMonthlyTemps.rotate!(6)
+    end
+  end
+
+  def calc_deep_ground_temperatures(runner, hpxml)
+    # Return deep annual ground temperature.
+    # Annual average ground temperature using Xing's model.
+    if !hpxml.nil?
+      has_gshp = false
+      hpxml.buildings.each do |hpxml_bldg|
+        has_gshp = true if hpxml_bldg.heat_pumps.select { |h| h.heat_pump_type == HPXML::HVACTypeHeatPumpGroundToAir }.size > 0
+      end
+      return if !has_gshp
+    end
+
+    deep_ground_temperatures = File.join(File.dirname(__FILE__), 'data', 'Xing_okstate_0664D_13659_Table_A-3.csv')
+    if not File.exist?(deep_ground_temperatures)
+      fail 'Could not find Xing_okstate_0664D_13659_Table_A-3.csv'
+    end
+
+    require 'csv'
+    require 'matrix'
+
+    v1 = Vector[header.Latitude, header.Longitude]
+    dist = 1 / Constants.small
+    temperatures_amplitudes = nil
+    CSV.foreach(deep_ground_temperatures) do |row|
+      v2 = Vector[row[3].to_f, row[4].to_f]
+      new_dist = (v1 - v2).magnitude
+      if new_dist < dist
+        temperatures_amplitudes = row[5..9].map(&:to_f)
+        dist = new_dist
+      end
+    end
+
+    tol = 3
+    if dist > tol # default values; not close enough
+      runner.registerWarning("Could not find nearby (#{dist.round(3)} > #{tol}) values from Xing's ground temperature model; using default values.")
+      # Following are from E+ docs
+      data.DeepGroundAnnualTemp = UnitConversions.convert(11.1, 'C', 'F')
+      data.DeepGroundSurfTempAmp1 = UnitConversions.convert(13.4, 'deltac', 'deltaf')
+      data.DeepGroundSurfTempAmp2 = UnitConversions.convert(0.7, 'deltac', 'deltaf')
+      data.DeepGroundPhaseShiftTempAmp1 = 25.0 # days
+      data.DeepGroundPhaseShiftTempAmp2 = 30.0 # days
+      return
+    end
```

Comment:
Everything looks good in the PR but now I'm questioning this.

First of all, these are _not_ default values. I looked it up. The E+ docs simply include an example and the example clearly shows that the values correspond to Chicago. There are no default values in E+, and using Chicago as a default is a poor choice.

Second, I think the original motivation of resorting to a default was to handle non-US locations.

Since we now handle 4000 locations internationally, I would think it'd nearly always be better to simply choose the closest location.

Reply:
Do we still want to register a warning when matching a location farther than 3?

Reply:
I tend to feel that we shouldn't bother.